test/ts/preact.tsx

```diff
@@ -387,13 +399,15 @@ h('form', { onSubmit: onSubmit });
 
 // Should accept onToggle
 const onToggle = (e: h.JSX.TargetedToggleEvent<HTMLDetailsElement>) => {};
```

Comment:
Shouldn't this be HTMLDialogElement?

Reply:
Er yes, yes it should. I think that was a typo.

On the other hand, it does show that our `h` and `createElement` types aren't quite correct as they'll accept handlers with generics for other elements than their own. IIUC, the way to fix this is add a `h` & `createElement` [type definition for every single element](https://github.com/preactjs/preact/blob/daba00255753b6c7764ca2649dd183163458a5f5/src/index.d.ts#L190-L200).